test/CMakeLists.txt

```diff
@@ -27,27 +27,22 @@ set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
 set(INSTALL_GTEST OFF CACHE BOOL "" FORCE)
 set(INSTALL_GMOCK OFF CACHE BOOL "" FORCE)
 
-find_package(GTest)
-if(NOT GTest_FOUND)
-  message(STATUS "GTest not found, downloading from source")
-  include(FetchContent)
-  FetchContent_Declare(googletest
-    GIT_REPOSITORY https://github.com/google/googletest.git
-    GIT_TAG        v1.14.0
-    GIT_SHALLOW    TRUE
-    GIT_PROGRESS   TRUE
-    OVERRIDE_FIND_PACKAGE
-  )
-  FetchContent_MakeAvailable(googletest)
-endif()
+include(FetchContent)
+FetchContent_Declare(googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG        v1.14.0
+  GIT_SHALLOW    TRUE
+  GIT_PROGRESS   TRUE
+  FIND_PACKAGE_ARGS NAMES GTest gtest
+)
+FetchContent_MakeAvailable(googletest)
 
-if(NOT TARGET GTest::GTest)
-    add_library(GTest::GTest ALIAS gtest)
-    add_library(GTest::Main ALIAS gtest_main)
+if(NOT TARGET GTest::gtest_main)
+    add_library(GTest::gtest_main ALIAS gtest_main)
```

Comment:
`GTest::Main` may not exist, but `GTest::gtest_main` should exist for every version.